corehq/change_ownership.py

```diff
@@ -0,0 +1,202 @@
+        user_data = user.get_user_data(DOMAIN)
+        if user_data[USER_TYPE_PROP_NAME] != 'rc':
+            write_to_log([user.user_id], is_success=False, message="User Type not RC")
+            skip_count += 1
+            continue
+
+        try:
+            # Get a descendant of user location which has the same rc number
+            loc = SQLLocation.objects.get(
+                domain=DOMAIN,
+                parent__location_id=user.location_id,
+                name=user_data[RC_NUM_PROP_NAME]
+            )
+            # Set this new descendant location as user location
+            user.location_id = loc.location_id
+            user_to_save.append(user)
+            write_to_log([user.user_id])
+            success_count += 1
+        except SQLLocation.DoesNotExist:
+            write_to_log(
+                [user.user_id],
+                is_success=False,
+                message=f'({user_data[RC_NUM_PROP_NAME]}) does not exist as child of location with id ({loc.location_id})'
+            )
+            fail_count += 1
+        finally:
+            # Time logging
+            end_time = time.time()
+            time_diff = end_time - start_time
+            total_time += time_diff
+            print(f'Time to process User #{idx}: {time_diff}s')
+            print(f'Estimate time remaining: {time_diff * (user_count - idx)}s')
+
+        print("Processing Users Complete!")
+        print(
+            f"Success: {success_count}, 
+            Fail: {fail_count}, 
+            Skipped: {skip_count},
+            Total Time: {round(total_time / 60, 2)} minutes"
+        )
+    CommCareUser.bulk_save(user_to_save)
+    print("Saving Users Complete!")
+
+### Task 2 & 3
+def transfer_case_ownership():
+    print("---MOVING CASE OWNERSHIP---")
+    case_ids = (
+        CaseSearchES()
+        .domain(DOMAIN)
+        .OR(
+            case_type('menage'),
+            case_type('membre'),
+            case_type('seance_educative'),
+            case_type('fiche_pointage')
+        )
+        .sort('opened_on')  # sort so that we can continue
+    ).get_ids()
+
+    case_count = len(case_ids)
+    batch_count = math.ceil(case_count / BATCH_SIZE)
+    print(f'Total Cases to Process: {case_count}')
+    print(f'Total Batches to Process: {batch_count}')
+    case_blocks = []
+    current_chunk = 0
+    end_time = total_time = 0
+    success_count = fail_count = 0
+    start_time = time.time()
+    for case_obj in CommCareCase.objects.iter_cases(case_ids, domain=DOMAIN):
+        try:
+            user = CommCareUser.get_by_user_id(case_obj.opened_by)
+        except CommCareUser.AccountTypeError as e:
+            write_to_log([case_obj.case_id], is_success=False, message='Not a mobile worker account')
+            fail_count += 1
+            continue
+        else:
+            case_block = CaseBlock(
+                create=False,
+                case_id=case_obj.case_id,
+                owner_id=user.location_id,
```

Comment:
Should we avoid update if its already this owner?

Reply:
This is a good point. We could probably log this separately and then simply ignore for updating.

Reply:
I've implemented the same for users, so that we don't save user updates if they are already at the correct location ([bf3b264](https://github.com/dimagi/commcare-hq/pull/34588/commits/bf3b264f740a226213cd2df2fbe86f6fd632fcb8))